### PR TITLE
fix APIs: cast, concat, tensor_array_to_tensor, argmin, argmax, argsort

### PR DESCRIPTION
**Pull request by zhupengyang**
- cast
image

- concat
image

- tensor_array_to_tensor
image

- argmin
image

- argmax
image

- argsort
image

### CODE REVIEW

File: python/paddle/fluid/layers/tensor.py [hunk 1]
```diff
@@ -147,23 +147,45 @@ def create_global_var(shape,
...
-    it to the output with :attr:`dtype`. It's meaningless if the output
-    dtype equals the input dtype, but it's fine if you do so.
+    This OP takes in the Variable :attr:`x` with :attr:`x.dtype` and casts it
+    to the output with :attr:`dtype`. It's meaningless if the output dtype
+    equals the input dtype, but it's fine if you do so.
 
     Args:
-        x (Variable): The input Variable for casting.
-        dtype(np.dtype|core.VarDesc.VarType|str): Data type of the output Variable.
+        x (Variable): An input N-D Tensor  with data type bool, float16,
```
Comment (hong19860320) [inline python/paddle/fluid/layers/tensor.py]: An input N-D Tensor  with data 多了个空格吧

File: python/paddle/fluid/layers/tensor.py [hunk 2]
```diff
@@ -147,23 +147,45 @@ def create_global_var(shape,
 
 def cast(x, dtype):
     """
-    This layer takes in the Variable :attr:`x` with :attr:`x.dtype` and casts
-    it to the output with :attr:`dtype`. It's meaningless if the output
-    dtype equals the input dtype, but it's fine if you do so.
+    This OP takes in the Variable :attr:`x` with :attr:`x.dtype` and casts it
```
Comment (hong19860320) [inline python/paddle/fluid/layers/tensor.py:152]: This OP还是The OP？

File: python/paddle/fluid/layers/tensor.py [hunk 3]
```diff
@@ -147,23 +147,45 @@ def create_global_var(shape,
...
-        x (Variable): The input Variable for casting.
-        dtype(np.dtype|core.VarDesc.VarType|str): Data type of the output Variable.
+        x (Variable): An input N-D Tensor  with data type bool, float16,
+            float32, float64, int32, int64, uint8.
+        dtype(np.dtype|core.VarDesc.VarType|str): Data type of the output:
+            bool, float15, float32, float64, int8, int32, int64, uint8.
 
     Returns:
-        Variable: The output Variable after casting.
+        Variable: A Tensor with the same shape as input.
```
Comment (cyj1986) [inline python/paddle/fluid/layers/tensor.py]: A Tensor-> A tensor,     as input -> as input's

File: python/paddle/fluid/layers/tensor.py [hunk 4]
```diff
@@ -180,27 +202,47 @@ def concat(input, axis=0, name=None):
...
+        axis(int, optional): Axis to compute indices along. The effective range
+            is [-R, R), where R is Rank(x). when axis<0, it works the same way
+            as axis+R. Default is 0.
+        name (str, optional): The default value is None. Normally there is no
+            need for user to set this property. For more information, please
+            refer to :ref:`api_guide_Name`.
 
     Returns:
-        Variable: Output variable of the concatenation
+        Variable: A tensor with the same data type as input.
```
Comment (cyj1986) [inline python/paddle/fluid/layers/tensor.py]: as input's

File: python/paddle/fluid/layers/tensor.py [hunk 5]
```diff
@@ -214,47 +256,48 @@ def concat(input, axis=0, name=None):
...
+            is [-R, R), where R is Rank(x). when axis<0, it works the same way
+            as axis+R. Default is 1.
+        name (str, optional): The default value is None. Normally there is no
+            need for user to set this property. For more information, please
+            refer to :ref:`api_guide_Name`.
 
     Returns:
-        Variable: Output variable of the concatenation
-        Variable: The input LodTensorArray items' dims along the axis
+        Variable: A LoDTensor with the same data type with input
```
Comment (cyj1986) [inline python/paddle/fluid/layers/tensor.py]: A LoDTensor  which has the same data type as input's

File: python/paddle/fluid/layers/tensor.py [hunk 6]
```diff
@@ -214,47 +256,48 @@ def concat(input, axis=0, name=None):
...
-
-        output_index.data = [3, 1, 2]
+    This OP concatenates the input LodTensorArray along the axis.
 
     Args:
-        input(list): Input LodTensorArray
-        axis(int): Integer axis along which the tensors will be concatenated
-        name(str|None): A name for this layer(optional). If set None, the layer
-                       will be named automatically.
+        input(list): A LodTensorArray with data type float32, float64, int32,
```
Comment (cyj1986) [inline python/paddle/fluid/layers/tensor.py]: 请确认这个input是list么

File: python/paddle/fluid/layers/tensor.py [hunk 7]
```diff
@@ -551,44 +646,70 @@ def argmax(x, axis=0):
...
+    :attr:`input`.
 
     Args:
-        input(Variable): The input Variable for sorting.
-        axis(int): The axis along which to sort the input Variable. When
-                   :attr:`axis` < 0, the actual axis will be :attr:`axis` +
-                   rank(:attr:`input`). Default -1, the last dimension.
-        name(str|None): (optional) A name for this layer. If set None, the
-                   layer will be named automatically.
+        x(Variable): An input N-D Tensor with type float32, float64, int16,
```
Comment (cyj1986) [inline python/paddle/fluid/layers/tensor.py]: 上面的函数参数，没有x, 是input 请确认

File: python/paddle/fluid/layers/tensor.py (hunk 3 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/tensor.py]: Done

File: python/paddle/fluid/layers/tensor.py (hunk 4 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/tensor.py]: Done

File: python/paddle/fluid/layers/tensor.py (hunk 5 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/tensor.py]: Done

File: python/paddle/fluid/layers/tensor.py (hunk 6 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/tensor.py]: 是Variable。Done

File: python/paddle/fluid/layers/tensor.py (hunk 7 above)
Comment (zhupengyang) [reply python/paddle/fluid/layers/tensor.py]: Done

**Review by cyj1986 — APPROVED** [review]

**Review by heavengate — APPROVED** [review]

**Pull request merged**